core/src/main/kotlin/io/customer/android/core/di/SDKComponent.kt

```diff
@@ -1,14 +1,21 @@
 package io.customer.android.core.di
 
+import io.customer.android.core.environment.BuildEnvironment
+import io.customer.android.core.environment.DefaultBuildEnvironment
 import io.customer.android.core.module.CustomerIOModule
+import io.customer.android.core.util.LogcatLogger
+import io.customer.android.core.util.Logger
 
 /**
  * Object level DiGraph for the SDK. Provides a centralized way to manage all
  * dependencies in the SDK in a single place.
  * The object can be accessed from anywhere in the SDK and can be used to provide
  * a consistent and efficient way to access them throughout the SDK.
  */
+@Suppress("MemberVisibilityCanBePrivate", "unused")
 object SDKComponent : DiGraph() {
     val androidSDKComponent: AndroidSDKComponent? get() = getOrNull()
+    val buildEnvironment: BuildEnvironment get() = singleton { DefaultBuildEnvironment() }
+    val logger: Logger get() = singleton { LogcatLogger(buildEnvironment = buildEnvironment) }
```

Comment:
Curious about the usage of `singleton` here?

- Since the `object` itself is a singleton, any class instances that it contains as members are also created once and persist for the lifetime of the application, effectively behaving as part of the singleton. Why are we adding another singleton on top of that?

Reply:
To make sure our tests don't break. I think the [earlier PR that introduced `singleton`](https://github.com/customerio/customerio-android/pull/307) could explain better about its usage.

Reply:
In the earlier PR, the `SDKComponent` didn't have `buildEnvironment` or `logger`. My questions was mostly around why did we go reusing the instance of these classes and not creating a newer one? 

 For example, `DefaultBuildEnvironment()` is just returning a static config. It will be accessed once as well?

Reply:
Because we used `singleton` here, both `Logger` and `BuildEnvironment` will be created only once, regardless of how many times they are accessed. We could use `newInstance` for `BuildEnvironment` if we prefer, but I don’t think it will have much effect.